nunaliit2-js/src/main/js/nunaliit2/n2.widgetLegend.js

```diff
@@ -28,558 +28,1306 @@ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+					} else if( 'n2_found' === key ){
+						context[key] = false;
+					} else {
+						context[key] = value;
+					};
+				};
+
+				var symbolizer = style.getSymbolizer(context);
+
+				// SVG
+				var svg = this._createSVGNode('svg');
+				if( svg ) {
+					this._setAttr(svg, 'version', '1.1');
+					this._setAttr(svg, 'viewBox', '-7 -7 14 14');
+					this._addClass(svg, 'n2widgetLegend_svg');
+					var $svg = $(svg);
+
+					// Geometry
+					var graphicName = symbolizer.getSymbolValue('graphicName',context);
+					var geom = null;
+					if( graphicName 
+							&& this.cachedSymbols[graphicName] ){
+						geom = this._createSVGNode('path');
+						this._setAttr(geom, 'd', this.cachedSymbols[graphicName]);
+
+					} else if( graphicName 
+							&& OpenLayers.Renderer.symbol[graphicName] ) {
+						var path = this._computePathFromSymbol(OpenLayers.Renderer.symbol[graphicName]);
+						this.cachedSymbols[graphicName] = path;
+						geom = this._createSVGNode('path');
+						this._setAttr(geom, 'd', this.cachedSymbols[graphicName]);
+
+					} else {
+						geom = this._createSVGNode('circle');
+						this._setAttr(geom, 'r', 5);
+					};
+					if( geom ) {
+						symbolizer.forEachSymbol(function(name,value){
+							if( 'r' === name ){
+								// Do not adjust radius
+							} else if( 'fill-opacity' === name ) {
+								// Make opacity more pronounced
+								var effectiveValue = (value * 0.5) + 0.5;
+								_this._setAttr(geom, name, effectiveValue);
+							} else {
+								_this._setAttr(geom, name, value);
+							};
+						},context);
+
+						svg.appendChild(geom);
+					};
+
+					$parent.append($svg);
+				};
+			},
+
+			_insertSvgPreviewLine: function($parent, style, context_){
+				var _this = this;
+
+				var context = {};
+				for(var key in context_){
+					var value = context_[key];
+
+					if( 'n2_hovered' === key ){
+						context[key] = false;
+					} else if( 'n2_selected' === key ){
+						context[key] = false;
+					} else if( 'n2_found' === key ){
+						context[key] = false;
+					} else {
+						context[key] = value;
+					};
+				};
+
+				var symbolizer = style.getSymbolizer(context);
+
+				// SVG
+				var svg = this._createSVGNode('svg');
+				if( svg ) {
+					this._setAttr(svg, 'version', '1.1');
+					this._setAttr(svg, 'viewBox', '-7 -7 14 14');
+					this._addClass(svg, 'n2widgetLegend_svg');
+					var $svg = $(svg);
+
+					// Geometry
+					var geom = this._createSVGNode('line');
+					this._setAttr(geom, 'x1', -5);
+					this._setAttr(geom, 'y1', 0);
+					this._setAttr(geom, 'x2', 5);
+					this._setAttr(geom, 'y2', 0);
+					if( geom ) {
+						symbolizer.forEachSymbol(function(name,value){
+							_this._setAttr(geom, name, value);
+						},context);
+
+						svg.appendChild(geom);
+					};
+
+					$parent.append($svg);
+				};
+			},
+
+			_insertSvgPreviewPolygon: function($parent, style, context_){
+				var _this = this;
+
+				var context = {};
+				for(var key in context_){
```

Comment:
This for loop to initialize the `context` variable is repeated a few times. Consider whether it could/should be moved into a function to avoid code duplication and improve clarity.